specklepy/objects/geometry.py

```diff
@@ -313,45 +563,146 @@ def _edge(self):
     def _curve_2d(self):
         return self._Brep.Curve2D[self.CurveIndex]
 
+    @classmethod
+    def from_list(cls, args: List[Any]) -> 'BrepTrim':
+        return cls(
+            EdgeIndex=args[0],
+            StartIndex=args[1],
+            EndIndex=args[2],
+            FaceIndex=args[3],
+            LoopIndex=args[4],
+            CurveIndex=args[5],
+            IsoStatus=args[6],
+            TrimType=BrepTrimTypeEnum(args[7]).name,
+            IsReversed=bool(args[8]),
+        )
+
+    def to_list(self) -> List[Any]:
+        encoded = []
+        encoded.append(self.EdgeIndex)
+        encoded.append(self.StartIndex)
+        encoded.append(self.EndIndex)
+        encoded.append(self.FaceIndex)
+        encoded.append(self.LoopIndex)
+        encoded.append(self.CurveIndex)
+        encoded.append(self.IsoStatus)
+        encoded.append(getattr(BrepTrimTypeEnum, self.TrimType).value)
+        encoded.append(self.IsReversed)
+        return encoded
+
 
 class Brep(
     Base,
     speckle_type=GEOMETRY + "Brep",
     chunkable={
-        "Surfaces": 200,
-        "Curve3D": 200,
-        "Curve2D": 200,
-        "Vertices": 5000,
+        "SurfacesValue": 200,
+        "Curve3DValues": 200,
+        "Curve2DValues": 200,
+        "VerticesValue": 5000,
         "Edges": 5000,
         "Loops": 5000,
-        "Trims": 5000,
+        "TrimsValue": 5000,
         "Faces": 5000,
     },
     detachable={"displayValue"},
+    serialize_ignore={"Surfaces", "Curve3D", "Curve2D", "Vertices", "Trims"},
 ):
     provenance: str = None
     bbox: Box = None
     area: float = None
     volume: float = None
     displayValue: Mesh = None
-    Surfaces: List[Surface] = None
-    Curve3D: List[Base] = None
-    Curve2D: List[Base] = None
-    Vertices: List[Point] = None
-    Edges: List[BrepEdge] = None
-    Loops: List[BrepLoop] = None
-    Trims: List[BrepTrim] = None
-    Faces: List[BrepFace] = None
+    SurfacesValue: List[float] = None
+    Curve3DValues: List[float] = None
+    Curve2DValues: List[float] = None
+    VerticesValue: List[float] = None
+    TrimsValue: List[float] = None
     IsClosed: bool = None
     Orientation: int = None
 
-    def __setattr__(self, name: str, value: Any) -> None:
-        if not value:
-            return
-        if name in {"Edges", "Loops", "Trims", "Faces"}:
-            for val in value:
-                val._Brep = self
-        super().__setattr__(name, value)
+    def _inject_self_into_children(self, children: Optional[List[Base]]) -> List[Base]:
+        if children is None:
+            return children
+
+        for child in children:
+            child._Brep = self
+        return children
+
+    @property
+    def Edges(self) -> List[BrepEdge]:
```

Comment:
as mentioned in the `DataChunk` comments, chunking should not be exposed to the users at all. similarly, it should be the `Edges`, `Loops`, `Trims`, etc that are defined on the class and the `{abc}Values` that have getters and setters since the `Values` lists are just for the benefit of serialisation / deserialisation.

eg `Curve3D` is defined on the class and is what the user interacts with when building and inspecting a brep
`Curve3DValues` has:
- a getter that returns `Curve3D` as a flat list of numbers (which the serialiser will chunk / dechunk as it does any other list)
- a setter that takes a flat list, decodes it to a list of curves, and sets that to `Curve3D`

Reply:
I'm torn about this change. I see where you're coming from with regards to exposing the parameters users interact with and obfuscating the stuff they shouldn't be worrying about. I also appreciate why we would prefer having `Curve3D` shown in the dataclass attributes and have the `@property` and `@Curve3DValue.setter` for the `Curve3DValues` attribute.

That being said, from a practical stand point it doesn't change anything for users. The Brep class will still expose `Curve3D` and `Curve3DValues` as public attributes (ie: attributes that don't start with `_`). If we were still using `pydantic` I would propose making `Curve3DValues` a private attribute like `_Curve3DValues` and use the `ModelField` `alias` attribute to ensure that (de)serialization occurs as expected.

Regardless, I'll change them just now as it makes sense from a style/linting perspective. It's worth bearing in mind though that the `Edges`, `Loops` and `Faces` will still have to be set using the `getter` and `setter` method so we can inject the `_brep` into them. Or maybe now is a good time to ask why we need to inject the `_brep` into them? It's pretty recursive and feels a bit dodgy.

Reply:
I resolved this in 5bc6b8c. Let me know what you think :raised_hands: